lib/ansible/executor/task_executor.py

```diff
@@ -772,13 +772,23 @@ def _execute_internal(self, templar: TemplateEngine, variables: dict[str, t.Any]
         # also now add connection vars results when delegating
         if self._task.delegate_to:
             result["_ansible_delegated_vars"] = {'ansible_delegated_host': self._task.delegate_to}
+
+            # TODO: consider deprecating or adding a better mechanism to pass
+            # variables to the results side. The comment above isn't really
+            # true since cvars is not templated.
```

Comment:
I think it is fine to pass them w/o templating, so they can examine the sources 'as they were', what we should also do is pass the 'resolved' options with the connection itself

also we should look to deprecate callbacks that rely on variables directly, but we should ensure that the interface for getting the resolved information from the connection object is working. 

in any case, the current fix seems to be fine, specially for backporting, but we want to get defined objects/interfaces for callbacks going forward.

Reply:
I'm not sure if there's a need. I overlooked that these are filtered out here https://github.com/ansible/ansible/blob/devel/lib/ansible/executor/task_result.py#L202, and after looking through the ansible-collections org, I don't see any callback plugins using anything other than "ansible_host" in the delegated variables in the ansible-collections org.

How does https://github.com/ansible/ansible/pull/85397 look for backporting? I removed the PlayContext changes.